### PR TITLE
Added background image support for sections

### PR DESCRIPTION
**Pull request by sdmcraft**
- [x] New variations introduced in this PR

Variation Name : Section with background  

- [x] Added to sidekick library

Test URLs:
- Original: https://www.sunstar.com/<path>
- Before: https://main--sunstar--hlxsites.hlx.live/_drafts/satyam/section-with-background
- After: https://section-bg--sunstar--hlxsites.hlx.live/_drafts/satyam/section-with-background

### CODE REVIEW

**Review by RitwikSrivastava — APPROVED** [review]

File: scripts/scripts.js [hunk 1]
```diff
@@ -300,6 +300,45 @@ export function addTopSpacingStyleToFirstMatchingSection(main) {
   });
 }
 
+function getViewPort() {
+  const { width } = getWindowSize();
+  if (width >= 1232) {
+    return 'dekstop';
```
Comment (dnbute) [inline scripts/scripts.js:306]: Nit: Typo; Accidentally this still works since it's the default in the switch below

**Review by dnbute — COMMENTED** [review]
I think we need to also add a height to the picture/section, this is the first thing I see when I open the test url

Screenshot 2023-11-07 at 09 41 50

**Comment by RitwikSrivastava** [comment]
Shouldnt we want the content to be inside the background image in case of tablets and mobile too ? 
Screenshot 2023-11-07 at 3 17 05 PM

**Comment by sdmcraft** [comment]
Please refresh the page if you are changing the viewport using browser dev tools as the image does not auto-change on viewport change.

**Comment by sdmcraft** [comment]
> I think we need to also add a height to the picture/section, this is the first thing I see when I open the test url

Since I am using images as background (in CSS) , they cannot control the size of the section of which they are background of. Need to see what other options we have...

**Comment by sdmcraft** [comment]
> Shouldnt we want the content to be inside the background image in case of tablets and mobile too ?

If the content is more than the image can accommodate, it'd spill out, reason being that enlarging the image to cover the whole section may end up showing an undesirable enlarged version of the image. I think the authors need to ensure that they use the right combination of background image and text i.e. use larger images if they have a lot of other content in a section.

**Comment by dnbute** [comment]
> > I think we need to also add a height to the picture/section, this is the first thing I see when I open the test url
> 
> Since I am using images as background (in CSS) , they cannot control the size of the section of which they are background of. Need to see what other options we have...

Could we do the same as the live page? Aka add a height to the section itself? It could be based on the background picture's dimensions

**Comment by sdmcraft** [comment]
> Could we do the same as the live page? Aka add a height to the section itself? It could be based on the background picture's dimensions

Yes, I changed the approach now to use an absolutely positioned `picture` element . This way the height of the picture is taken into consideration. Please take a look at the test url now.

**Review by dnbute — APPROVED** [review]
lgtm

**Review by RitwikSrivastava — APPROVED** [review]

**Pull request merged**